### PR TITLE
Case study 1 in Python

### PR DESCRIPTION
**Pull request by taodav**
In this pull request is a Python implementation of the environment for the first case study (Sec. 4.1) in the JellyBean World paper, as per #10. 

This new environment is registered under `gym` as `JBW-v1` and `JBW-render-v1` for the non-rendering and rendering version of the environment respectively. So to use this environment, simply call
```
import gym
import jbw

env = gym.make('JBW-v1')
```

NOTE: there are some slight discrepancies between the configuration mentioned in the paper vs. what was done in the swift implementation. Referencing #10, this implementation uses values from the swift implementation as opposed to what's listed in the paper.

On top of this, I've also included a small change in the Makefile for the visualization - I've referenced the linked libraries after the files to compile due to a referencing issue.

### CODE REVIEW

File: jbw/visualizer/Makefile [hunk 1]
```diff
@@ -84,7 +84,7 @@ $(BIN_DIR)/%.spv: %.spv
 	cp $< $@
 
 jbw_visualizer: bin $(LIBS) $(JBW_VISUALIZER_OBJS) $(JBW_VISUALIZER_SHADERS:%=$(BIN_DIR)/%)
-		$(CPP) -o $(BIN_DIR)/jbw_visualizer $(CPPFLAGS) $(LDFLAGS) $(JBW_VISUALIZER_OBJS)
+		$(CPP) -o $(BIN_DIR)/jbw_visualizer $(JBW_VISUALIZER_OBJS) $(CPPFLAGS) $(LDFLAGS)
```
Comment (asaparov) [inline jbw/visualizer/Makefile:87]: Just curious, why did you need to change the order of arguments here?

**Review by asaparov — APPROVED** [review]
This looks good for the first case study!

To implement the other case studies, it would be helpful to implement the Reward Functions and Schedules described at the bottom of page 7 in the paper (implemented in Swift [here](https://github.com/eaplatanios/jelly-bean-world/blob/master/api/swift/Sources/JellyBeanWorld/Rewards.swift)). I don't think `gym.make` allows you to pass custom reward functions, so we should probably just register a new gym environment for each case study.

File: jbw/visualizer/Makefile (hunk 1 above)
Comment (taodav) [reply jbw/visualizer/Makefile:87]: I was running into weird issues with compiling the visualizers when the linked libraries were before referenced before the cpp file itself - something about an undefined reference?

Comment (asaparov) [reply jbw/visualizer/Makefile:87]: Oh right, I think libraries are supposed to be specified after object files. I'll change that elsewhere too.
